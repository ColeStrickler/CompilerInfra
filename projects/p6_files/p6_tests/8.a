func : (b: bool) -> int {
	a : int = 4;
    a++;
    a--;

    a = a/4;
    a = a*4;
    a = a+4;

    x : bool;
    toconsole -a;
    x = !b;
    return a;
}




main : (x: int, z : bool) -> void {

    myVar : int = func(z); 
    x = myVar - 7;
}