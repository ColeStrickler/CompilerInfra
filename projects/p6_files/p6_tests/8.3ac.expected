[BEGIN GLOBALS]
func
main
[END GLOBALS]
[BEGIN <init> LOCALS]
[END <init> LOCALS]
fun_<init>: enter <init>
lbl_0:      leave <init>
[BEGIN func LOCALS]
b (formal arg of 8 bytes)
a (local var of 8 bytes)
x (local var of 8 bytes)
tmp0 (tmp var of 8 bytes)
tmp1 (tmp var of 8 bytes)
tmp2 (tmp var of 8 bytes)
tmp3 (tmp var of 8 bytes)
tmp4 (tmp var of 8 bytes)
[END func LOCALS]
fun_func:   enter func
            getarg 1 [b]
            [a] := 4
            [a] := [a] ADD64 1
            [a] := [a] SUB64 1
            [tmp0] := [a] DIV64 4
            [a] := [tmp0]
            [tmp1] := [a] MULT64 4
            [a] := [tmp1]
            [tmp2] := [a] ADD64 4
            [a] := [tmp2]
            [tmp3] := NEG64 [a]
            WRITE [tmp3]
            [tmp4] := NOT64 [b]
            [x] := [tmp4]
            setret [a]
            goto lbl_1
lbl_1:      leave func
[BEGIN main LOCALS]
x (formal arg of 8 bytes)
z (formal arg of 8 bytes)
myVar (local var of 8 bytes)
tmp0 (tmp var of 8 bytes)
tmp1 (tmp var of 8 bytes)
[END main LOCALS]
main:       enter main
            getarg 1 [x]
            getarg 2 [z]
            setarg 1 [z]
            call func
            getret [tmp0]
            [myVar] := [tmp0]
            [tmp1] := [myVar] SUB64 7
            [x] := [tmp1]
lbl_2:      leave main